daemon/containerd/image_changes.go

```diff
@@ -0,0 +1,69 @@
+package containerd
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/containerd/containerd/content"
+	"github.com/containerd/containerd/mount"
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/google/uuid"
+	"github.com/opencontainers/image-spec/identity"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/sirupsen/logrus"
+)
+
+func (i *ImageService) Changes(ctx context.Context, container *container.Container) ([]archive.Change, error) {
+	cs := i.client.ContentStore()
+
+	imageManifestBytes, err := content.ReadBlob(ctx, cs, *container.ImageManifest)
+	if err != nil {
+		return nil, err
+	}
+	var manifest ocispec.Manifest
+	if err := json.Unmarshal(imageManifestBytes, &manifest); err != nil {
+		return nil, err
+	}
+
+	imageConfigBytes, err := content.ReadBlob(ctx, cs, manifest.Config)
+	if err != nil {
+		return nil, err
+	}
+	var image ocispec.Image
+	if err := json.Unmarshal(imageConfigBytes, &image); err != nil {
+		return nil, err
+	}
+
+	rnd, err := uuid.NewRandom()
+	if err != nil {
+		return nil, err
+	}
+
+	snapshotter := i.client.SnapshotService(i.snapshotter)
```

Comment:
I guess this is also one of those places where we still need to decide if we want to have the _container_ as source of truth for the snapshotter to use;

```suggestion
	snapshotter := i.client.SnapshotService(container.Driver)
```

But we can look at those "as a whole" and change them all at once.

Reply:
Yeah, there's been a few fragmented discussions about this, maybe we can bring it up in one of the maintainers calls and get some input/come to a decision about what we'd like to do. For now I'll leave this one like this.